lib/rubocop/cop/lint/unneeded_disable.rb

```diff
@@ -24,6 +24,9 @@ def check(offenses, cop_disabled_line_ranges, comments)
               comment = comments.find { |c| c.loc.line == line_range.begin }
               unneeded_cop = find_unneeded(comment, offenses, cop, cop_offenses,
                                            line_range)
+
+              return nil if unneeded_cop == COP_NAME
```

Comment:
I didn't realize before, but now I think this solution is too simple. The `Lint/UnneededDisable` cop might be enabled again further down in the file, so it's probably necessary to compare ranges rather than bailing out.